tests/test_java.py

```diff
@@ -215,10 +231,20 @@ def sap_and_crash(self, *args, **kwargs):
         return result
 
     monkeypatch.setattr(AsyncProfiledProcess, "start_async_profiler", sap_and_crash)
-    # To make sure it is reverted to True (the original value) after the test
-    monkeypatch.setattr(JavaProfiler, "_should_profile", True)
 
-    with JavaProfiler(1, 5, Event(), str(tmp_path), False, False, "cpu", 0, False, "ap") as profiler:
+    profiler = JavaProfiler(
```

Comment:
Why did you change the arguments here?

Reply:
I needed to change `False` to `'all'` now. Also added kwargs for clarity

Reply:
Isn't `False` supposed to be the same as `''`? `all` is the "new" `True` right?

Reply:
Yeah but this conversion happens in the `argparse` / `configparse` level. `JavaProfiler` now expects this value as a string, so I had to change it.